libs/form-elements/src/NumberInput.tsx

```diff
@@ -115,14 +115,23 @@ export default function NumberInput({
 }: NumberInputProps) {
   const intlContext = useIntlContext();
   const id = `numberformat-${name}`;
+  const usesDecimalSeparator = decimalSeparator !== undefined;
+  const usesThousandSeparator = thousandSeparator !== undefined;
 
-  if (!intlContext && (!decimalSeparator || !thousandSeparator)) {
+  if (!intlContext && (!usesDecimalSeparator || !usesThousandSeparator)) {
     throw new Error(
-      "You must pass decimalSeparator and thousandSeparator props if you are using the NumberInput component without IntlProvider.",
+      "You must pass decimalSeparator and/or thousandSeparator props if you are using the NumberInput component without IntlProvider.",
     );
   }
+
+  if (usesDecimalSeparator && decimalSeparator === "") {
```

Comment:
add .trim to check for a whitespace decimal separators. With this solution someone could send `'\t'`